docs/explanation/testing.md

```diff
@@ -70,21 +118,26 @@ Integration tests typically take significantly longer to run than unit tests.
 * Upgrade sequence
   * Regression test: upgrade stable/candidate/beta/edge from charmhub with the locally-built charm.
 
-
 ```{caution}
 
 When writing an integration test, it is not sufficient to simply check that Juju reports that running the action was successful; rather, additional checks need to be executed to ensure that whatever the action was intended to achieve worked.
 
 ```
 
-**Tools.**
+### Tools
 
 - [`pytest`](https://pytest.org/) and/or [`unittest`](https://docs.python.org/3/library/unittest.html) and
 - [pytest-operator](https://github.com/charmed-kubernetes/pytest-operator) and/or [`zaza`](https://github.com/openstack-charmers/zaza)
```

Comment:
I think we should drop mention of zaza (I and I think most of our team knows nothing about it):

```suggestion
- [pytest-operator](https://github.com/charmed-kubernetes/pytest-operator)
```